scanner/mappers/mappers.go

```diff
@@ -693,16 +718,11 @@ func cvssVector(cvssVector string) (severityValues, error) {
 	return values, nil
 }
 
-func nvdSeverityAndScores(vuln *claircore.Vulnerability, nvdVulns map[string]*nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
+func nvdSeverityAndScores(vuln *claircore.Vulnerability, v *nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
 	values := severityValues{
 		severity: vuln.Severity,
 	}
 
-	v, ok := nvdVulns[vuln.ID]
```

Comment:
we should have a `nil` check here, as it's possible this is `nil`

Reply:
Do you mean where we read `nvdVulns` map? We only add entries there if the list of CVE is non-empty.

Reply:
I meant we should make sure we check

```
if v == nil {
	return values, errors.New("cannot find NVD data")
}
```

Reply:
going to resolve this, as at the moment, we call this via `&nvdVuln`, so we know right now `v != nil`, but I think we should future-proof this, in case we make changes to how it's called

Reply:
> going to resolve this, as at the moment, we call this via &nvdVuln

I understand now what you meant in the original comment.

Checking for nill and returning an error would make us skip the severity information and potentially ignore the programming mistake. You want to balance nil checks vs. handling programming errors, such as passing a nil pointer inadvertently to a standard library function.